### ISSUE TITLE
[FEATURE] Add BiVAE to the benchmark results

### ISSUE THREAD

**Issue by tqtg** [microsoft/recommenders#1362]
### Description

Hi,

I would like to add the benchmark results for the BiVAE model. I just wonder if the benchmark notebook is open for contribution?

Thanks.

**Comment by gramhagen** [comment]
Yeah, it's definitely open, please just make sure you use the same data prep/splitting approach (seeds) so the comparisons are as close as possible.

**Comment by tqtg** [comment]
Just one question. When I submit a new PR, do I need to re-run the notebook for all algorithms, or do I just need to make sure the newly added algorithm is runnable?

**Comment by gramhagen** [comment]
I think it's preferable if the entire notebook is re-run.

**Comment by tqtg** [comment]
I see that the number of EPOCHS is fixed to 15. Is it a hard constraint?
Some of the algorithms may find it difficult to converge after 15 epochs. 

In this particular case, the BiVAE model can not converge after 15 epochs. It's not the same as other deterministic models that we can increase the learning rate for faster convergence, BiVAE is a probabilistic model which requires sampling process during training. Thus, with small number of epochs, it cannot gather enough samples to learning good representations for users and items.

Any chance that we can let number of epochs parameter model-dependent?

**Comment by miguelgfierro** [comment]
yes agree, the most important thing is to have a fair representation, so we use the same dataset in all and use the same split so we are sure we have the same data

In terms of the epochs, I see in the BiVAE notebook, you are using 500 epochs and only takes 1 min or so. I think this is ok. I'm looking [at the table now](https://github.com/microsoft/recommenders/blob/main/examples/06_benchmarks/movielens.ipynb), maybe we could do 15 epochs for NCF, fastai, LightGCN, 50 epochs for SVD, ALS and BPR and 500 for BiVAE? 

@tqtq, do you think these numbers are reasonable?

Also, what we can do, after you do the PR, we can use one of our VMs to have the final results, right now we did the test on  Standard NC6 (6 vcpus, 55 GB memory, 1K80 GPU), we can use that or another one more modern.

**Comment by tqtg** [comment]
It's a bit difficult to say what should be the right number of epochs. Given that it was set to 15, I guess the contributors had already tried to tune other hyper-params of their models (e.g., learning_rate, latent_factors, batch_size) to obtain reasonably good results.

If we agree that the number of epochs can be model-dependent, I think we can let the contributors know that they can update their settings if they wish to. On my side, I will take care of BPR and BiVAE as those are the ones I put in.

@gramhagen @miguelgfierro is that good?

**Comment by miguelgfierro** [comment]
sounds good
